src/workerd/api/pyodide/pyodide.h

```diff
@@ -282,71 +282,78 @@ struct MemorySnapshotResult {
 // CPU architecture-specific artifacts. The logic for loading these is in getArtifacts.
 class ArtifactBundler: public jsg::Object {
  public:
-  kj::Maybe<const PyodidePackageManager&> packageManager;
-  // ^ lifetime should be contained by lifetime of ArtifactBundler since there is normally one worker set for the whole process. see worker-set.h
-  // In other words:
-  // WorkerSet lifetime = PackageManager lifetime and Worker lifetime = ArtifactBundler lifetime and WorkerSet owns and will outlive Worker, so PackageManager outlives ArtifactBundler
-  kj::Maybe<MemorySnapshotResult> storedSnapshot;
-
-  ArtifactBundler(kj::Maybe<const PyodidePackageManager&> packageManager,
-      kj::Maybe<kj::Array<const kj::byte>> existingSnapshot,
-      bool isValidating = false)
-      : packageManager(packageManager),
-        storedSnapshot(kj::none),
-        existingSnapshot(kj::mv(existingSnapshot)),
-        isValidating(isValidating) {};
+  struct State: public kj::Refcounted {
```

Comment:
Why refcounted? workerd does not require multiple references to this but the internal repo does in a couple of places.